### PR TITLE
Wizard recipe: mold-v1.0.0

### PR DESCRIPTION
**Pull request by miguelraz**
This pull request contains a new build recipe I built using the BinaryBuilder.jl wizard:

* Package name: mold
* Version: v1.0.0

@staticfloat please review and merge.

### CODE REVIEW

File: M/mold/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+sources = [
+    GitSource("https://github.com/rui314/mold.git", "a9f9163d2cad5f83d6865b736c2797cf49bde387")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd mold/
+make -j${nproc}
+make install
```
Comment (giordano) [inline M/mold/build_tarballs.jl:18]:
```suggestion
make install PREFIX="${prefix}" BINDIR="${bindir}" LIBDIR="${libdir}"
```

File: M/mold/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd mold/
+make -j${nproc}
+make install PREFIX="${prefix}" BINDIR="${bindir}" LIBDIR="${libdir}"
+mv mold ${WORKSPACE}/destdir/
+exit
```
Comment (giordano) [inline M/mold/build_tarballs.jl:20]:
```suggestion
```

File: M/mold/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+# The products that we will ensure are always built
+products = [
+    ExecutableProduct("mold", :mold)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="Zlib_jll", uuid="83775a58-1f1d-513f-b197-d71354ab007a"))
+    Dependency(PackageSpec(name="xxHash_jll", uuid="5fdcd639-92d1-5a06-bf6b-28f2061df1a9"))
+    Dependency(PackageSpec(name="OpenSSL_jll", uuid="458c3c95-2e84-50aa-8efc-19380b2a3a95"))
```
Comment (giordano) [inline M/mold/build_tarballs.jl:45]: Why does a linker depend on OpenSSL? :confused:

File: M/mold/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6", preferred_gcc_version = v"11.1.0")
```
[the remark this answers is not in the document]
Comment (miguelraz) [reply M/mold/build_tarballs.jl:49]: It uses C++20, I tried it with the previous ones and failed.

[the remark this answers is not in the document]
Comment (giordano) [reply M/mold/build_tarballs.jl:49]: Why is everybody using C++20 this week?

[the remark this answers is not in the document]
Comment (miguelraz) [reply M/mold/build_tarballs.jl:49]: Safer concurrency and object lifetime abatractions.

File: M/mold/build_tarballs.jl (hunk 3 above)
Comment (miguelraz) [reply M/mold/build_tarballs.jl:45]: SHA256: https://twitter.com/rui314/status/1471994301232349188?s=21

**Comment by giordano** [comment]
Edit the file?

**Comment by miguelraz** [comment]
@giordano Is this good to merge?

**Comment by giordano** [comment]
If you wanted macOS in not really, since you aren't building it

**Comment by miguelraz** [comment]
Just reread the manual - MacOs will be supported in the 2.0 release, this should be good for linux for now.

**Pull request merged**